reconstructSPI/iterative_refinement/expectation_maximization.py

```diff
@@ -371,17 +380,19 @@ def generate_slices(map_3d_f, xy_plane, n_pix, rots):
             of projection of rotated map_3d_f.
             Shape (n_rotations, n_pix, n_pix)
         xyz_rotated : arr
-            Rotated xy plane.
-            Shape (n_pix**2, 3)
+            Rotated xy planes.
+            Shape (n_rotations, 3, n_pix**2)
         """
         n_rotations = rots.shape[0]
-        # map_values interpolation, calculate from map, rots
-        map_3d_f = np.ones_like(map_3d_f)
-        xyz_rotated = np.ones_like(xy_plane)
+        slices = np.empty((n_rotations, map_3d_f.shape[0], map_3d_f.shape[1]))
+        xyz_rotated = np.empty((n_rotations, n_pix**2, 3))
+        for i in range(n_rotations):
+            xyz_rotated[i] = rots[i] @ (xy_plane + 0.5) - 0.5
```

Comment:
you need to match the rotation on the axis of the xy_plane with 3 dimensions for the xyz coords. You are doing this in `rots[i] @ (xy_plane + 0.5) - 0.5` so that should work.

put in some documentation about this +0.5, -0.5. perhaps in `test_generate_slices` 

however, the shape of `xyz_rotated = np.empty((n_rotations, n_pix**2, 3))` doesn't match what is needed in `xyz_rotated[i] = rots[i] @ (xy_plane + 0.5) - 0.5`.

this should work: `xyz_rotated = np.empty((n_rotations, 3, n_pix**2))`

Reply:
Yup, honest mistake there as I made that change late. 

We might not need to hop on a call after all - you hit the nail on the head with your interpretation of the +0.5 change - that was what I was working on getting solved in https://github.com/compSPI/reconstructSPI/pull/40#discussion_r841327505

Reply:
Notes from call with @thisTyler:

The +/-0.5 is not strictly needed, we would just get a slice with an one pixel stripe of empty values (zeros) for a 180 deg rotation _after using map_coordinates_.

 However, think that the generate_slices should not have that behaviour as it is not expected.
 
 Since `generate_slices` returns the xyz_rotated, which will be used in `insert_slices`, and these coordinates will be used in the same way with `map_coordinates`, we are free to to the +/- 0.5 thing to have more intuitive behaviour.